Revert poetry.lock and pyproject.toml to version on master

dragon-dxw opened the pull request:
We encountered a problem where a Promo Group containing at least 1 item would cause a 500 error:

```
File /usr/lib/python3.9/site-packages/wagtailnhsukfrontend/templatetags/nhsukfrontend_tags.py:82 in chunk
return [input_list[i:i + size] for i in range(0, len(input_list), size)]
File /usr/lib/python3.9/site-packages/wagtailnhsukfrontend/templatetags/nhsukfrontend_tags.py:82 in <listcomp>
return [input_list[i:i + size] for i in range(0, len(input_list), size)]
File /usr/lib/python3.9/site-packages/wagtail/core/blocks/list_block.py:85 in __getitem__ args locals
return self.bound_blocks[i].value
AttributeError: 'list' object has no attribute 'value'
```

We reverted most of the changes back (except for the removed migration core/0047 and the actual migration core/0048, and an update to how poetry itself is installed) and got it working, but that didn't apply all the changes we'd wanted to make.

If this works, we'll have a sensible setup from which to try to understand the underlying reasons for this error. (It works nicely locally.)

dragon-dxw committed:
Revert poetry.lock and pyproject.toml to version on master

We encountered a problem where a Promo Group containing at least 1
item would cause a 500 error:

```
File /usr/lib/python3.9/site-packages/wagtailnhsukfrontend/templatetags/nhsukfrontend_tags.py:82 in chunk
return [input_list[i:i + size] for i in range(0, len(input_list), size)]
File /usr/lib/python3.9/site-packages/wagtailnhsukfrontend/templatetags/nhsukfrontend_tags.py:82 in <listcomp>
return [input_list[i:i + size] for i in range(0, len(input_list), size)]
File /usr/lib/python3.9/site-packages/wagtail/core/blocks/list_block.py:85 in __getitem__ args locals
return self.bound_blocks[i].value
AttributeError: 'list' object has no attribute 'value'
```

We reverted most of the changes back (except for the removed migration
core/0047 and the actual migration core/0048, and an update to how
poetry itself is installed) and got it working, but that didn't apply all
the changes we'd wanted to make.

If this works, we'll have a sensible setup from which to try to understand
the underlying reasons for this error. (It works nicely locally.)

jkeasley reviewed (approved):
This looks good to me

dragon-dxw committed:
Require non-broken ast >= 1.5.0

Black fails in CI:

```
Creating nhsx-website_test_run ... done
Traceback (most recent call last):
...
  File "/usr/lib/python3.9/site-packages/black/__init__.py", line 52, in <module>
    from typed_ast import ast3, ast27
  File "/usr/lib/python3.9/site-packages/typed_ast/ast3.py", line 40, in <module>
    from typed_ast import _ast3
ImportError: Error relocating /usr/lib/python3.9/site-packages/typed_ast/_ast3.cpython-39-x86_64-linux-musl.so: _PyUnicode_DecodeUnicodeEscape: symbol not found
```

https://stackoverflow.com/questions/69912264/python-3-9-8-fails-using-black-and-importing-typed-ast-ast3
https://github.com/python/typed_ast/issues/169#issuecomment-966711709
